src/secrets.ts

```diff
@@ -0,0 +1,58 @@
+import { HDKey } from '@scure/bip32';
+import { generateMnemonic, validateMnemonic, mnemonicToSeedSync } from '@scure/bip39';
+import { wordlist } from '@scure/bip39/wordlists/english';
+import { encodeBase64toUint8 } from './base64';
+import { bytesToNumber } from './utils';
+import { hexToNumber } from '@noble/curves/abstract/utils';
+export const generateNewMnemonic = (): string => {
+	const mnemonic = generateMnemonic(wordlist, 128);
+	if (!validateMnemonic(mnemonic, wordlist)) {
+		return generateNewMnemonic();
+	}
+	return mnemonic;
+};
+
+export const deriveSeedFromMnemonic = (mnemonic: string): Uint8Array => {
+	const seed = mnemonicToSeedSync(mnemonic);
+	return seed;
+};
+
+export const deriveSecret = (seed: Uint8Array, keysetId: string, counter: number): Uint8Array => {
+	return derive(seed, keysetId, counter, 0);
```

Comment:
use a const to make it clear what `0` represents

Reply:
very good. I was looking at the code and was like wrf is this `0`